### PR TITLE
fix for default AD role. During user login-in, if his role in AD has …

### PR DESCRIPTION
**Pull request by ibilous**
…changed but his XM role is not in configured AD group then he will stay with existing role

### CODE REVIEW

File: src/main/java/com/icthh/xm/uaa/security/ldap/UaaLdapUserDetailsContextMapper.java [hunk 1]
```diff
@@ -96,12 +98,7 @@ private String mapRole(TenantProperties.Ldap.Role roleConf, Collection<? extends
                 }
             });
         }
-        if (mappedRoles.isEmpty()) {
-            log.info("Role mapping not found. Default role {} will be used", mappedXmRole);
-        } else {
-            mappedXmRole = mappedRoles.getLast();
-        }
-
+        String mappedXmRole = mappedRoles.isEmpty() ? null : mappedRoles.getLast();
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/security/ldap/UaaLdapUserDetailsContextMapper.java:101]: i think better return Optional than null

@amedvedchuk

Comment (ibilous) [reply src/main/java/com/icthh/xm/uaa/security/ldap/UaaLdapUserDetailsContextMapper.java]: fixed

**Review by sergeysenja1992 — APPROVED** [review]

File: src/test/java/com/icthh/xm/uaa/security/ldap/UaaLdapUserDetailsContextMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    @InjectMocks
+    UaaLdapUserDetailsContextMapper uaaLdapUserDetailsContextMapper;
+
+    @Test
+    public void userRoleMustNotChangeDuringLogin() {
+        whenFindUserByLogin();
+        when(ldapConf.getRole()).thenReturn(new TenantProperties.Ldap.Role("DEFAULT_ROLE", null));
+        uaaLdapUserDetailsContextMapper.mapUserFromContext(ctx, LOGIN, List.of());
+        verify(userDetailsService).loadUserByUsername(eq(LOGIN));
+        verify(userService, never()).saveUser(any(User.class));
```
Comment (amedvedchuk) [inline src/test/java/com/icthh/xm/uaa/security/ldap/UaaLdapUserDetailsContextMapperTest.java:60]: is it enough to verify that we never save the user or better to make a specific check that role still the same?

Comment (ibilous) [reply src/test/java/com/icthh/xm/uaa/security/ldap/UaaLdapUserDetailsContextMapperTest.java:62]: added additional check

**Review by ishkurko — APPROVED** [review]

**Pull request merged**